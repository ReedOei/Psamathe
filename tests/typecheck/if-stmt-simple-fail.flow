contract Main {}
;

if true = 1 { pass } else { pass }

