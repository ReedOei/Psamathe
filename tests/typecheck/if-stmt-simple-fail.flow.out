  <generatedTop>
    <common>
      <k>
        1 : ! bool ~> assertEnv ( .Map ) ~> assertEnv ( .Map ) ~> pass ~> assertEnv ( ?Delta ) ~> setEnv ( .Map ) ~> pass ~> assertEnv ( ?Xi ) ~> setEnv ( join ( ?Delta , ?Xi ) ) ~> .
      </k>
      <contracts>
        SetItem ( Main )
      </contracts>
      <fields>
        Main |-> .Set
      </fields>
      <constructors>
        Main |-> constructor ( .VarDefs , pass )
      </constructors>
      <functions>
        Main |-> .Set
      </functions>
      <types>
        Main |-> SetItem ( type ( Main , SetItem ( asset )
        SetItem ( consumable ) , Main ) )
        address |-> SetItem ( type ( address , SetItem ( consumable )
        SetItem ( nonfungible ) , address ) )
        bool |-> SetItem ( type ( bool , SetItem ( consumable )
        SetItem ( nonfungible ) , bool ) )
        nat |-> SetItem ( type ( nat , SetItem ( consumable )
        SetItem ( fungible ) , nat ) )
        string |-> SetItem ( type ( string , SetItem ( consumable )
        SetItem ( nonfungible ) , string ) )
      </types>
      <freshCounter>
        0
      </freshCounter>
    </common>
    <typeEnv>
      .Map
    </typeEnv>
  </generatedTop>
#And
  {
    ?Gamma0
  #Equals
    .Map
  }
#And
  {
    ?Gamma
  #Equals
    .Map
  }
#And
  {
    ?Tau
  #Equals
    ! bool
  }
